src/NCronJob/Scheduler/JobProcessor.cs

```diff
@@ -33,7 +33,10 @@ public async Task ProcessJobAsync(JobRun jobRun, CancellationToken cancellationT
 
             await jobExecutor.RunJob(jobRun, cancellationToken).ConfigureAwait(false);
 
-            jobRun.NotifyStateChange(JobStateType.Completed);
+            if (!jobRun.IsCompleted)
```

Comment:
Without this, a faulted job (for instance) would eventually be marked as Completed.